### PR TITLE
json: implement copy_strings=false

### PR DESCRIPTION
**Pull request by xackus**

### CODE REVIEW

File: lib/std/json.zig [hunk 1]
```diff
@@ -1376,11 +1376,9 @@ pub const Parser = struct {
     }
 
     fn parseString(p: *Parser, allocator: *Allocator, s: std.meta.TagPayloadType(Token, Token.String), input: []const u8, i: usize) !Value {
-        // TODO: We don't strictly have to copy values which do not contain any escape
-        // characters if flagged with the option.
         const slice = s.slice(input, i);
         switch (s.escapes) {
-            .None => return Value{ .String = try mem.dupe(allocator, u8, slice) },
+            .None => return Value{ .String = if (p.copy_strings) try mem.dupe(allocator, u8, slice) else slice },
```
Comment (daurnimator) [inline lib/std/json.zig]: This means that strings with escapes in them are copied, and those without escapes are not. The caller would have no indication of that fact, so how would they free them?

File: lib/std/json.zig [hunk 2]
```diff
@@ -1613,3 +1608,36 @@ test "escaped characters" {
...
+    const input =
+        \\{
+        \\  "noescape": "aą😂",
+        \\  "simple": "\\\/\n\r\t\f\b\"",
+        \\  "unicode": "\u0105",
+        \\  "surrogatepair": "\ud83d\ude02"
+        \\}
+    ;
+
+    const tree_nocopy = try Parser.init(debug.global_allocator, false).parse(input);
```
Comment (andrewrk) [inline lib/std/json.zig]: avoid use of `debug.global_allocator` in new test cases, this is causing the OutOfMemory. try this:

```zig
var mem_buf: [1000]u8 = undefined;
const allocator = &std.heap.FixedBufferAllocator.init(&mem_buf).allocator;
```

File: lib/std/json.zig (hunk 1 above)
Comment (andrewrk) [reply lib/std/json.zig]: If you assume an arena allocator then you don't need such an indication. Does the implementation set up an arena allocator? (Genuine question; I'm typing this on a phone and don't remember )

Comment (xackus) [reply lib/std/json.zig]: It does use an arena allocator.

File: lib/std/json.zig (hunk 2 above)
Comment (xackus) [reply lib/std/json.zig]: Is the state of the debug allocator kept between tests?

Comment (andrewrk) [reply lib/std/json.zig]: yes, it's global. the main use case was to be an easy allocator to use for temporary tests or one-off things. but then of course it ends up getting used. so now it's fully deprecated and will be removed

**Comment by andrewrk** [comment]
nice work

**Pull request merged**